components/raftstore/src/store/fsm/apply.rs

```diff
@@ -958,21 +974,22 @@ where
         if !data.is_empty() {
             let cmd = util::parse_data_at(data, index, &self.tag);
 
-            if apply_ctx.yield_high_latency_operation && has_high_latency_operation(&cmd) {
+            if has_high_latency_operation(&cmd) && apply_ctx.yield_high_latency_operation {
                 self.priority = Priority::Low;
-                if apply_ctx.priority != Priority::Low {
-                    apply_ctx.commit(self);
-                    return ApplyResult::Yield;
-                }
             }
-            if should_write_to_engine(&cmd) || apply_ctx.kv_wb().should_write_to_engine() {
+            let need_sync = should_write_to_engine(&cmd)
+                && self.last_sync_apply_index != self.apply_state.get_applied_index();
```

Comment:
When does `last_sync_apply_index` equal to `self.apply_state.get_applied_index()`?

Reply:
After all the changes in `kv_wb` and `pending_sst` are write to db,  `last_sync_apply_index` will be equal to `self.apply_state.get_applied_index()`, see also  [`commit_opt`](https://github.com/gozssky/tikv/blob/batch-ingest/components/raftstore/src/store/fsm/apply.rs#L464).

Reply:
Oh, I understand it. The code here just for not writing twice for the same cmd?

Reply:
Not really. The code here is mainly for ensuring all changes before this command are flushed, because we may bypass `kv_wb` to write data, for example ingest sst directly.